src/acom_music_box/music_box.py

```diff
@@ -458,10 +463,16 @@ def solve(self, path_to_output = None):
         headers.append("ENV.temperature")
         headers.append("ENV.pressure")
 
+        #music_box_logger.progress("solve0152 self.solver = {}".format(self.solver))
+
+        if (self.solver is None):
+            music_box_logger.progress("Warning: MusicBox object {} has no solver."
+                .format(self))
```

Comment:
I think this is a failure. We shouldn't ever reach here, but maybe we raise an exception after the logged statement, and maybe make it critical instead of progres

Reply:
If someone writing an alternate main() forgets to create a solver, then the failure will occur within the Fortran code where a Python programmer will have a hard time diagnosing it. Yes, we can throw an exception here instead of sending a known **None** solver into Fortran.

Reply:
Just an additional note: if self.solver is None, then the following line of python code:
        rate_constant_ordering = musica.user_defined_reaction_rates(self.solver)

terminates the process without any message or warning or stack trace. I will raise the exception to provide some diagnostic information . . .